cpp/src/io/orc/stripe_enc.cu

```diff
@@ -1409,9 +1409,9 @@ void decimal_sizes_to_offsets(device_2dspan<rowgroup_rows const> rg_bounds,
   if (rg_bounds.count() == 0) return;
   std::vector<decimal_column_element_sizes> h_sizes;
   h_sizes.reserve(elem_sizes.size());
-  for (auto& [col_idx, current_sizes] : elem_sizes) {
-    h_sizes.emplace_back(col_idx, current_sizes);
-  }
+  std::transform(elem_sizes.begin(), elem_sizes.end(), std::back_inserter(h_sizes), [](auto& p) {
+    return decimal_column_element_sizes{p.first, p.second};
+  });
```

Comment:
This seems even less elegant to me.

Reply:
yeah, but it follows the "no raw loops" guideline :)

Reply:
The for each style is also "no raw loop"  though. As long as you don't use any explicit index, it is good.

Reply:
(in theory) transform is better as it's a less general tool.
I'm not defending this piece for code specifically, talking about the guideline.
Should I revert the change?

Reply:
No need for it. They are about the same, you didn't have to changed to this, and don't have to change back :P

Reply:
Glad we agree :D